include/mbedtls/sha256.h

```diff
@@ -59,7 +59,17 @@ typedef struct mbedtls_sha256_context {
 mbedtls_sha256_context;
 
 #else  /* MBEDTLS_SHA256_ALT */
+
```

Comment:
I think all alt headers are included under `extern "C"`. We should be consistent: sha256 shouldn't be different from the others.